cxx/emissions/categorical.hh

```diff
@@ -20,14 +20,9 @@ class CategoricalEmission : public Emission<int> {
     }
   };
 
-  void incorporate(const std::pair<int, int>& x) {
-    ++N;
-    emission_dists[x.first].incorporate(x.second);
-  }
-
-  void unincorporate(const std::pair<int, int>& x) {
```

Comment:
Should unincorporated be deleted?

Reply:
Yes.  The base.hh Distribution class defines unincorporate(x) to just be incorporate(x, -1.0) and that definition is totally appropriate here.